modules/navigation/godot_navigation_server.cpp

```diff
@@ -899,37 +997,79 @@ COMMAND_2(obstacle_set_radius, RID, p_obstacle, real_t, p_radius) {
 	obstacle->set_radius(p_radius);
 }
 
+real_t GodotNavigationServer::obstacle_get_radius(RID p_obstacle) const {
+	NavObstacle *obstacle = obstacle_owner.get_or_null(p_obstacle);
+	ERR_FAIL_NULL_V(obstacle, 0);
+
+	return obstacle->get_radius();
+}
+
 COMMAND_2(obstacle_set_height, RID, p_obstacle, real_t, p_height) {
 	NavObstacle *obstacle = obstacle_owner.get_or_null(p_obstacle);
 	ERR_FAIL_NULL(obstacle);
 	obstacle->set_height(p_height);
 }
 
+real_t GodotNavigationServer::obstacle_get_height(RID p_obstacle) const {
+	NavObstacle *obstacle = obstacle_owner.get_or_null(p_obstacle);
+	ERR_FAIL_NULL_V(obstacle, 0);
+
+	return obstacle->get_height();
+}
+
 COMMAND_2(obstacle_set_velocity, RID, p_obstacle, Vector3, p_velocity) {
 	NavObstacle *obstacle = obstacle_owner.get_or_null(p_obstacle);
 	ERR_FAIL_NULL(obstacle);
 
 	obstacle->set_velocity(p_velocity);
 }
 
+Vector3 GodotNavigationServer::obstacle_get_velocity(RID p_obstacle) const {
+	NavObstacle *obstacle = obstacle_owner.get_or_null(p_obstacle);
+	ERR_FAIL_NULL_V(obstacle, Vector3());
+
+	return obstacle->get_velocity();
+}
+
 COMMAND_2(obstacle_set_position, RID, p_obstacle, Vector3, p_position) {
 	NavObstacle *obstacle = obstacle_owner.get_or_null(p_obstacle);
 	ERR_FAIL_NULL(obstacle);
 	obstacle->set_position(p_position);
 }
 
+Vector3 GodotNavigationServer::obstacle_get_position(RID p_obstacle) const {
+	NavObstacle *obstacle = obstacle_owner.get_or_null(p_obstacle);
+	ERR_FAIL_NULL_V(obstacle, Vector3());
+
+	return obstacle->get_position();
+}
+
 void GodotNavigationServer::obstacle_set_vertices(RID p_obstacle, const Vector<Vector3> &p_vertices) {
 	NavObstacle *obstacle = obstacle_owner.get_or_null(p_obstacle);
 	ERR_FAIL_NULL(obstacle);
 	obstacle->set_vertices(p_vertices);
 }
 
+Vector<Vector3> GodotNavigationServer::obstacle_get_vertices(RID p_obstacle) const {
+	NavObstacle *obstacle = obstacle_owner.get_or_null(p_obstacle);
+	ERR_FAIL_NULL_V(obstacle, dummy_empty_vertices);
```

Comment:
I think with the ref removed this `dummy_empty_vertices` can now be `Vector<Vector3>()` as it is more familiar and commonly used across the engine in similar cases.

Reply:
Yup, I agree, thanks for the catch!